### PR TITLE
Wrap in List

### LINKED ISSUE #5419: Wrap in List!
Related to #5442

### PR DESCRIPTION
**Pull request by liady**
**Problem:**
We cannot currently wrap with a List

**Fix:**
This PR fixes wrapping a single element with a list.

**Details:**
1. Adding a List as an option for "Wrap..."
2. Adjust the code so that wrapping in a map (which is currently two `INSERT`s in disguise) will actually `REPLACE` the map contents
3. NOTE - this PR supports a single element wrapping.

For multiple element wrapping see #5696

<video src="https://github.com/concrete-utopia/utopia/assets/7003853/0dfc0062-8c9a-4e19-9bfc-5a138f0a2df3"></video>

- [X] I opened a hydrogen project and it loaded
- [X] I could navigate to various routes in Preview mode

Fixes #5419

### CODE REVIEW

File: editor/src/components/editor/store/insertion-path.ts [hunk 1]
```diff
@@ -68,6 +68,23 @@ export function childInsertionPath(
   }
 }
 
+export interface MapInsertionPath {
+  type: 'MAP_INSERTION'
+  intendedParentPath: StaticElementPath
+  insertBehavior: ConditionalClauseInsertBehavior
```
Comment (Rheeseyb) [inline editor/src/components/editor/store/insertion-path.ts:74]: This type should probably be renamed

Comment (bkrmendy) [reply editor/src/components/editor/store/insertion-path.ts:74]: for the note, we agreed not to go with `MapInsertionPath` at all

File: editor/src/core/model/element-metadata-utils.ts [hunk 2]
```diff
@@ -2318,6 +2318,15 @@ export const MetadataUtils = {
     const element = MetadataUtils.findElementByElementPath(metadata, target)
     return MetadataUtils.isJSXMapExpressionFromMetadata(element)
   },
+  isJSXElementFromMetadata(element: ElementInstanceMetadata | null): boolean {
```
Comment (Rheeseyb) [inline editor/src/core/model/element-metadata-utils.ts:2321]: I'm surprised these two don't already exist, though we do have `getJSXElementFromMetadata` and `getJSXElementFromElementInstanceMetadata`

Comment (liady) [reply editor/src/core/model/element-metadata-utils.ts:2321]: yeah, I followed the `isConditionalFromMetadata`, `isJSXMapExpressionFromMetadata`, etc pattern here - should I leave it like this (for completion and for the next developer who will need it and look for it) or do you think I need to use the `getJSXElement...` methods?

**Review by bkrmendy — APPROVED** [review]

File: editor/src/core/model/element-metadata-utils.ts (hunk 2 above)
Comment (liady) [reply editor/src/core/model/element-metadata-utils.ts:2321]: so maybe we should leave this method for completion and for next developers, if that makes sense

Comment (Rheeseyb) [reply editor/src/core/model/element-metadata-utils.ts:2321]: Yeah, keep the function 👍

**Pull request merged**